pkg/tenant/db/organizations.go

```diff
@@ -0,0 +1,56 @@
+package db
+
+import (
+	"time"
+
+	"github.com/oklog/ulid/v2"
+	"github.com/rotationalio/ensign/pkg/utils/ulids"
+	"github.com/vmihailenco/msgpack/v5"
+)
+
+const OrganizationNamespace = "organizations"
+
+type Organization struct {
+	ID       ulid.ULID `msgpack:"id"`
+	Name     string    `msgpack:"name"`
+	Created  time.Time `msgpack:"created"`
+	Modified time.Time `msgpack:"modified"`
+}
+
+var _ Model = &Organization{}
+
+func (o *Organization) Key() (key []byte, err error) {
+	// Add check for null value
+	return o.ID.MarshalBinary()
+}
+
+func (o *Organization) Namespace() string {
+	return OrganizationNamespace
+}
+
+func (o *Organization) MarshalValue() ([]byte, error) {
+	return msgpack.Marshal(o)
+}
+
+func (o *Organization) UnmarshalValue(data []byte) error {
+	return msgpack.Unmarshal(data, o)
+}
+
+// VerifyOrg will check that resources are allocated to the correct organization.
+// The method will take in an orgID and will return true if the orgID of a resource
+// (tenant, member, project, topic, api key) is the same and an error if it is not.
+func VerifyOrg(orgID ulid.ULID, modelOrgID ulid.ULID) (bool, error) {
+	if ulids.IsZero(orgID) {
+		return false, ErrMissingOrgID
+	}
+
+	if ulids.IsZero(modelOrgID) {
+		return false, ErrMissingID
+	}
+
+	if orgID.Compare(modelOrgID) == 0 {
+		return true, nil
+	} else {
+		return false, ErrOrgNotVerified
+	}
+}
```

Comment:
What I was thinking we'd do here is take the orgID from the claims and the resourceID from the request (e.g. from the URL or the body of the request); fetch the orgID from the database using the resourceID then compare the claimsID and the orgID. 

```suggestion
func VerifyOrg(claimsOrgID ulid.ULID, resourceID ulid.ULID) (bool, error) {
	if ulids.IsZero(orgID) {
		return false, ErrMissingOrgID
	}

	if ulids.IsZero(resourceID) {
		return false, ErrMissingID
	}
	
	orgID, err := GetOrgIndex(resourceID)
	if err != nil {
	    return false, err 
	}

	if orgID.Compare(claimsOrgID) == 0 {
		return true, nil
	} else {
		return false, ErrOrgNotVerified
	}
}
```